src/GameState/DirectionControl.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { Pressable } from 'native-base';
+import { Entypo } from '@expo/vector-icons';
+import { MARBLE_SIZE } from './sizes';
+
+export const DirectionControl = (props: DirectionControlProps) => {
+  const handlePressDirection = (direction: string) => {
+    props.checkAndMoveMarble(direction);
+    props.setMarbleClickedCoordinates(undefined);
+  };
+
+  const getPressableInformation = (direction: string) => {
+    switch (props.direction) {
+      case 'N':
+        return {
+          left: `${props.coordinates.x * props.boxMarbleSize}%`,
+          top: `${
+            props.coordinates.y * props.boxMarbleSize - props.boxMarbleSize
+          }%`,
+          nameIcon: 'arrow-bold-up' as any,
+        };
+
+      case 'E':
+        return {
+          left: `${
+            props.coordinates.x * props.boxMarbleSize + props.boxMarbleSize
+          }%`,
+          top: `${props.coordinates.y * props.boxMarbleSize}%`,
+          nameIcon: 'arrow-bold-right' as any,
+        };
+
+      case 'S':
+        return {
+          left: `${props.coordinates.x * props.boxMarbleSize}%`,
+          top: `${
+            props.coordinates.y * props.boxMarbleSize + props.boxMarbleSize
+          }%`,
+          nameIcon: 'arrow-bold-down' as any,
+        };
+
+      case 'W':
+        return {
+          left: `${
+            props.coordinates.x * props.boxMarbleSize - props.boxMarbleSize
+          }%`,
+          top: `${props.coordinates.y * props.boxMarbleSize}%`,
+          nameIcon: 'arrow-bold-left' as any,
+        };
+    }
```

Comment:
I am not a big fan on switch case return. Moreover using a switch without a default value means that you do not need a switch. The way you calculate left and top are roughly similar in all cases.